generator/src/main/java/com/scottlogic/deg/generator/walker/rowspec/PotentialRowSpecCount.java

```diff
@@ -0,0 +1,45 @@
+package com.scottlogic.deg.generator.walker.rowspec;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.scottlogic.deg.generator.decisiontree.ConstraintNode;
+import com.scottlogic.deg.generator.decisiontree.DecisionNode;
+import com.scottlogic.deg.generator.decisiontree.DecisionTree;
+
+public class PotentialRowSpecCount {
+    private final int max;
+
+    @Inject
+    public PotentialRowSpecCount(@Named("config:internalRandomRowSpecStorage") int max) {
+        this.max = max;
+    }
+
+    public boolean lessThanMax(DecisionTree decisionTree){
```

Comment:
I'd suggest using `Optional<Integer>` here instead of relying on null having a meaningful behaviour.

This would allow a series of mappings, as `empty -> empty`, while `some(x) -> (some(f(x))` for any `f(x)`. It appears that's what we're trying to encode here.

Conversely, the entire logic is encapsulated within this class and not exposed, so it's really your call.